src/Network/Network/NatGateway/SetAzureRMNatGatewayCommand.cs

```diff
@@ -0,0 +1,125 @@
+﻿// <auto-generated>
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// 
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// 
+// 
+// Warning: This code was generated by a tool.
+// 
+// Changes to this file may cause incorrect behavior and will be lost if the
+// code is regenerated.
+// 
+// For documentation on code generator please visit
+//   https://aka.ms/nrp-code-generation
+// Please contact wanrpdev@microsoft.com if you need to make changes to this file.
+// </auto-generated>
+
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using System.Net;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using AutoMapper;
+using MNM = Microsoft.Azure.Management.Network.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+
+    [Cmdlet(VerbsCommon.Set, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NatGateway", SupportsShouldProcess = true, DefaultParameterSetName = "SetByNameParameterSet"), OutputType(typeof(PSNatGateway))]
+    public class SetAzureNatGatewayCommand : NetworkBaseCmdlet
+    {
+        private const string SetByNameParameterSet = "SetByNameParameterSet";
+        private const string SetByInputObjectParameterSet = "SetByInputObjectParameterSet";
+        private const string SetByResourceIdParameterSet = "SetByResourceIdParameterSet";
+
+        [Parameter(Mandatory = true, ParameterSetName = SetByNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = SetByNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Nat Gateway Id",
+            ParameterSetName = SetByResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string NatGatewayId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
```

Comment:
Please update to `Manadatory = true,` once Parameter set below added. This defines Required/Optional state inside of parameter set only